src/mod_roster.erl

```diff
@@ -445,41 +439,39 @@ item_to_xml(Item) ->
     #xmlel{name = <<"item">>, attrs = Attrs4,
            children = SubEls}.
 
+-spec process_iq_set(jid:jid(), jid:jid(), jlib:iq()) -> jlib:iq().
 process_iq_set(#jid{lserver = LServer} = From, To, #iq{sub_el = SubEl} = IQ) ->
     #xmlel{children = Els} = SubEl,
     mongoose_hooks:roster_set(LServer, ok, From, To, SubEl),
     lists:foreach(fun(El) -> process_item_set(From, To, El) end, Els),
     IQ#iq{type = result, sub_el = []}.
 
+-spec process_item_set(jid:jid(), jid:jid(), exml:element()) -> ok.
 process_item_set(From, To, #xmlel{attrs = Attrs} = El) ->
     JID1 = jid:from_binary(xml:get_attr_s(<<"jid">>, Attrs)),
     do_process_item_set(JID1, From, To, El);
```

Comment:
Could we rename `JID1` to something meaningful as we are already touching this code?

Reply:
Same as below, naming convention in this module is consistently terrible. I think that on the future PR modifying the hooks I could be discovering better naming conventions, for now it's a bit intractable I'm afraid.

Reply:
Ok, it's a must now, I just thought you already had them figured out.